### DISCUSSION TITLE
Question about initialize of shared memory

### DISCUSSION THREAD

**Discussion by advancenXX** [eProsima/Fast-DDS#2574] [Q&A] [answered]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

Do not want to alloc physical memory when init shm, something like copy-on-write.
Or, could we drop the [memset] line?

### Current behavior

        // Memset the whole segment to zero in order to force physical map of the buffer
        auto buffer = shared_mem_segment_->alloc_buffer(configuration_.segment_size(),
                        (std::chrono::steady_clock::now() + std::chrono::milliseconds(100)));
        memset(buffer->data(), 0, configuration_.segment_size());

### Steps to reproduce

App set segment_size to 8MB, then checking its RSS.

### Fast DDS version/commit

2.5.0

### Platform/Architecture

Ubuntu Focal 20.04 amd64

### Transport layer

Shared Memory Transport (SHM)

**Comment by JLBuenoLopez** [comment]
@advancenXX,

I am going to move this question to its proper forum according to our [CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md#contribution-guidelines).

**Comment by MiguelCompany** [comment, marked as answer]
@advancenXX This is by design, mainly due to the following reasoning:

1. Real-time behavior. We want determinism, so we should do the mapping as soon as possible. This way, all subsequent writes on the segment will have almost the same WCET.
2. Safety. We want the code to fail fast if the user configuration is not consistent with the physical limitations of the system. Doing the mapping as soon as possible ensures failing as close to the misconfiguration as possible.

**Reply by advancenXX** [reply]
Got it, thanks!
